src/devices/Ads1115/Ads1115.cs

```diff
@@ -62,37 +140,106 @@ public DataRate DataRate
-        /// <param name="dataRate">Data Rate</param>
-        public Ads1115(I2cDevice i2cDevice, InputMultiplexer inputMultiplexer = InputMultiplexer.AIN0, MeasuringRange measuringRange = MeasuringRange.FS4096, DataRate dataRate = DataRate.SPS128)
+        public DeviceMode DeviceMode
         {
-            _i2cDevice = i2cDevice;
-            _inputMultiplexer = inputMultiplexer;
-            _measuringRange = measuringRange;
-            _dataRate = dataRate;
+            get
+            {
+                return _deviceMode;
+            }
+            set
+            {
+                _deviceMode = value;
+                SetConfig();
+            }
+        }
 
-            SetConfig();
+        /// <summary>
+        /// Comparator mode.
+        /// Only relevant if the comparator trigger event is set up and is changed by <see cref="EnableComparator(short, short, ComparatorMode, ComparatorQueue)"/>.
+        /// </summary>
+        public ComparatorMode ComparatorMode
+        {
+            get;
+            private set;
         }
 
         /// <summary>
-        /// Set ADS1115 Config Register
+        /// Comparator polarity. Indicates whether the rising or the falling edge of the ALRT/RDY Pin is relevant.
+        /// Default: Low (falling edge)
+        /// </summary>
+        public ComparatorPolarity ComparatorPolarity
+        {
+            get
+            {
+                return _comparatorPolarity;
+            }
+            set
+            {
+                _comparatorPolarity = value;
+                SetConfig();
+            }
+        }
+
+        /// <summary>
+        /// Comparator latching mode. If enabled, the ALRT/RDY Pin will be kept signaled until the conversion value is read.
+        /// Only relevant when the comparator is enabled.
+        /// </summary>
+        public ComparatorLatching ComparatorLatching
+        {
+            get
+            {
+                return _comparatorLatching;
+            }
+            set
+            {
+                _comparatorLatching = value;
+                SetConfig();
+            }
+        }
+
+        /// <summary>
+        /// Minimum number of samples exceeding the lower/upper threshold before the ALRT pin is asserted.
+        /// This can only be set with <see cref="EnableComparator(short, short, ComparatorMode, ComparatorQueue)"/>.
+        /// </summary>
+        public ComparatorQueue ComparatorQueue
+        {
+            get
+            {
+                return _comparatorQueue;
+            }
+        }
+
+        /// <summary>
+        /// This event fires when a new value is available (in conversion ready mode) or the comparator threshold is exceeded.
+        /// Requires setup through <see cref="EnableConversionReady"/> or <see cref="EnableComparator(double, double, ComparatorMode, ComparatorQueue)"/>.
+        /// </summary>
+        public event Action AlertReadyAsserted;
```

Comment:
NIT: usually .NET events follow the naming pattern of `OnEventRaised` so this case it would probably be something like `OnNewAlertReadyAsserted` or somehting like that: [https://docs.microsoft.com/en-us/dotnet/csharp/event-pattern](https://docs.microsoft.com/en-us/dotnet/csharp/event-pattern)

Reply:
If I read that document correctly, that applies for the event handlers, not the names of the events themselves. 

It has 

```
void OnEventRaised(object sender, EventArgs args);
```
but 
```
public event EventHandler<FileFoundArgs> FileFound;
```

we're talking about the later case here.

Reply:
You are correct.